### PR TITLE
SAP2000_Toolkit: Bar Fragments

### LINKED ISSUE #42: SAP2000_Toolkit: ReadBar() to get properties as fragment
#### Description:

Get all available data from bar objects when reading:

- [x] Auto Mesh
- [ ] Direct Analysis Modifiers
- [x] Design Procedure
- [ ] End Skew
- [ ] Fireproofing
- [ ] Hinge Assigns
- [x] Insertion Point
- [ ] Lateral Bracing
- [ ] Material Overwrite
- [ ] Material Temperature
- [x] Modifiers
- [ ] Notional Size
- [ ] Output Stations
- [ ] TC Limits

### PR DESCRIPTION
**Pull request by maryannewachter**
### Issues addressed by this PR

Closes #42 for the following three features:
- [x] Auto Mesh
- [x] Design Procedure
- [x] Insertion Point - only location, not additional offsets.

Read/Write capabilities created for the features above.

### Test files

https://burohappold.sharepoint.com/:f:/r/sites/BHoM/02_Current/12_Scripts/01_Issue/BHoM/SAP2000_Toolkit/%2342-ReadBar-fragment-properties?csf=1&web=1&e=Jct68Q

### CODE REVIEW

File: SAP2000_Adapter/CRUD/Read/Bar.cs [hunk 1]
```diff
@@ -119,9 +122,77 @@ private List<Bar> ReadBars(List<string> ids = null)
...
+                        sectionModifier.Iz = sectionModifiers[4];
+                        sectionModifier.Iy = sectionModifiers[5];
+                        // mass modifier = 6
+                        // weight modifier = 7
+                        if (bhomBar.SectionProperty == null)
+                        {
+                            ISectionProperty sectionProperty = new ExplicitSection();
+                            bhomBar.SectionProperty = sectionProperty;
+                        }
+                        bhomBar.SectionProperty.Fragments.Add(sectionModifier);
```
Comment (JosefTaylor) [inline SAP2000_Adapter/CRUD/Read/Bar.cs:190]: SAP allows modifiers on the bar and on the section, vs. BHoM only allows them on the section. Section modifiers are already implemented in ReadSectionProperty(); this should not modify the section, but should create a modifier fragment on the bar with these values

Comment (JosefTaylor) [reply SAP2000_Adapter/CRUD/Read/Bar.cs:190]: Correction: modifiers on sections are not implemented- only modifiers on SurfaceProperties. I believe section modifiers have been implemented in ETABS.

File: SAP2000_Adapter/CRUD/Create/Bar.cs [hunk 2]
```diff
@@ -145,6 +152,114 @@ private bool SetObject(Bar bhBar)
...
+            }
+
+            if (m_model.FrameObj.SetInsertionPoint(name, (int)bhBar.InsertionPoint(), false, bhBar.ModifyStiffnessInsertionPoint(), ref offset1, ref offset2) != 0)
+            {
+                CreatePropertyWarning("Insertion point and perpendicular offset", "Bar", name);
+            }
+            
+            // Section Property Modifiers
+
+            if (bhBar.SectionProperty.FindFragment<SectionModifier>() != null)
```
Comment (JosefTaylor) [inline SAP2000_Adapter/CRUD/Create/Bar.cs:244]: same as below, this should be looking for modifiers on the bar, not on the section property

**Review by JosefTaylor — CHANGES_REQUESTED** [review]
Everything except for property modifiers works as expected and the code looks good.

File: SAP2000_Adapter/CRUD/Read/Bar.cs (hunk 1 above)
Comment (maryannewachter) [reply SAP2000_Adapter/CRUD/Read/Bar.cs:190]: Section modifiers are implemented in ETABS, but are on the section itself, not the bar.

File: SAP2000_Adapter/CRUD/Create/Bar.cs (hunk 2 above)
Comment (maryannewachter) [reply SAP2000_Adapter/CRUD/Create/Bar.cs]: Per discussion, removed section property modifiers till later PR.

**Comment by JosefTaylor** [comment]
image

**Review by JosefTaylor — APPROVED** [review]

**Comment by JosefTaylor** [comment]
@BHoMBot check compliance

**Comment by JosefTaylor** [comment]
Looks like we're only allowed filenames that are properly related to the method name:

> Method name must start with or end with the name of the file - For more information see [https://github.com/BHoM/documentation/wiki/MethodNameContainsFileName](https://github.com/BHoM/documentation/wiki/MethodNameContainsFileName)

**Comment by maryannewachter** [comment]
> Looks like we're only allowed filenames that are properly related to the method name:
> 
> > Method name must start with or end with the name of the file - For more information see https://github.com/BHoM/documentation/wiki/MethodNameContainsFileName
> 
> Also, some copy locals are set to true:
> 
> > Project reference for 'Adapter_Engine' should be set to not copy local
> 
> As well as Dimensional_oM and Analytical_oM.

Renamed files in Enums, Query, Fragments, and Modify to match method names to satisfy CI.

**Comment by maryannewachter** [comment]
@BHoMBot check compliance

**Comment by maryannewachter** [comment]
@BHoMBot check compliance

**Comment by maryannewachter** [comment]
@BHoMBot check compliance

**Comment by maryannewachter** [comment]
@BHoMBot check installer

**Review by enarhi — DISMISSED** [review]
_Reviewing for code compiling only, not tested for functionality_

LGTM

**Review by JosefTaylor — DISMISSED** [review]
LGTM

**Comment by JosefTaylor** [comment]
I'm about to rebase and force push this branch, if you have it cloned, please delete the branch locally. @enarhi @maryannewachter @jeanlucjackson

**Comment by JosefTaylor** [comment]
@BHoMBot check compliance

**Comment by JosefTaylor** [comment]
@BHoMBot check installer

**Review by JosefTaylor — APPROVED** [review]
Already approved

**Pull request merged**